### DISCUSSION TITLE
Downloading posts from a date range doesn't work.

### DISCUSSION THREAD

**Discussion by Zhiro90** [TumblThreeApp/TumblThree#400] [General]
After authenticating, I'm trying to archive my likes from the beginning of my tumblr  (2010~) to 2012~, but after messing around with the FROM/TO fields (with proper yyyymmdd format) it still downloads my most recent likes, with a few old likes thrown in.

I've tried 
FROM:(blank) TO:20120101,
FROM:20070606 TO:20120101
FROM:20110101 TO:20120101 

and all of them start downloading 2022 posts. 

Am I doing something wrong?

**Comment by thomas694** [comment]
There's a problem, because the chosen date is used for pagination, but Tumblr compares it against the date the post was liked, while TumblThree uses the date for filtering (post date).
We have to consider that and change the way TumblThree is doing it a bit.

What does a user expect from this start and end date here (ignoring the tooltip)? The date range the posts were liked or the date range the posts have been published? The latter somehow requires scanning always through all the liked posts.

**Reply by Zhiro90** [reply]
From /likedby/ requests i imagine people would expect the liked date. For regular tumblr requests i imagine post dates are expected. 

Still the posts I'm getting weren't liked nor created anywhere near 2022

**Reply by thomas694** [reply]
Please try the [test version](https://ci.appveyor.com/api/buildjobs/4sxnf3544wu2610w/artifacts/artifacts%2FTumblThree-v2.9.2-x64-Application.zip) and tell us, if it works.

**Reply by Zhiro90** [reply]
Same results,  these settings:

![Screenshot 2022-12-05 120512](https://user-images.githubusercontent.com/25686691/205713092-08c64daf-f535-4dd4-b7a3-6ac1df6c92e7.png)

Give me these downloads:

![Screenshot 2022-12-05 120615](https://user-images.githubusercontent.com/25686691/205713152-cc47930e-315f-440e-ad2d-061c72272aa4.png)

Both crawlers do the same thing

**Reply by thomas694** [reply]
In the beginning you said "archive my likes" (own likes) and in this example it shows "/liked/by" (other person's likes).
For downloading own likes (/likes) it should work. Maybe you can try it again with your own likes.

The "liked/by" downloader has a problem extracting the correct dates from the pages. Additionally, other person's likes don't have a date when the post was liked. We'll need to check this.

**Reply by Zhiro90** [reply]
Oh , but That's my blog. It requested me to authenticate to get the likes downloaded . What's the proper way to archive my own likes then? Thank you in advance 

Edit: you mentioned "/likes". I'll try that later

**Reply by thomas694** [reply]
You can download other people's shared likes (or your own shared) by using https://www.tumblr.com/liked/by/... But for your own likes you should use https://www.tumblr.com/likes as it's better.
The authentication is needed in both, because of the way we get the post details.
